backend/modules/simulation/router.py

```
@@ -182,3 +222,51 @@ async def get_user_progress(
     except Exception:
         logger.exception("Failed to get user progress", extra={"user_id": current_user.id, "user_progress_id": user_progress_id})
         raise HTTPException(status_code=500, detail="Internal server error")
+
+
+@router.get("/job/{job_id}/status")
+async def get_job_status_endpoint(
+    job_id: str,
+    current_user: User = Depends(get_current_user)
```

Ruff (0.14.8)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)